lib/rust/parser/src/syntax/token/operator.rs

```diff
@@ -267,50 +262,56 @@ impl HasOperatorProperties for variant::CommaOperator {
 /// Value that can be compared to determine which operator will bind more tightly within an
 /// expression.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Precedence {
-    /// A numeric value determining precedence order.
-    pub(super) value: u8,
+#[repr(u8)]
+#[allow(missing_docs)]
+pub enum Precedence {
+    /// A value that is lower than the precedence of any operator.
+    Min = 0,
+    Assignment,
+    TypeAnnotation,
+    Arrow,
+    Not,
+    Logical,
+    Equality,
+    Functional,
+    BitwiseOr,
+    BitwiseAnd,
+    Inequality,
+    Addition,
+    Multiplication,
+    Exponentiation,
+    OtherUserOperator,
+    Negation,
+    Application,
+    Annotation,
+    /// A value that is higher than the precedence of any operator.
+    Max,
 }
 
 impl Precedence {
     /// Return a precedence that is lower than the precedence of any operator.
     pub fn min() -> Self {
-        Precedence { value: 0 }
+        Precedence::Min
     }
 
-    /// Return the precedence for any operator.
+    /// Return the lowest precedence for any operator.
     pub fn min_valid() -> Self {
-        Precedence { value: 1 }
+        debug_assert_eq!(Precedence::Assignment as u8, Precedence::Min as u8 + 1);
```

Comment:
An assert that two `enum` values are next to each other!? Can it be false? Shouldn't it be apparent or a check to verify in a test and not code?